### PR TITLE
asana-power cli

### PR DESCRIPTION
**Pull request by kenankigunda**
Added asana-power command-line interface as an example demonstrating using the node-asana client library for power-user operations over the Asana API. 

See README.md for full usage notes

Corresponding Asana task: https://app.asana.com/0/0/1148193717192013/f

### CODE REVIEW

File: javascript/powercommands/command_runner.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const commands = {
```
Comment (kenankigunda) [inline javascript/powercommands/command_runner.js]: Code structure: consider making each command a class that knows how to parse/execute itself.

File: javascript/powercommands/task_client_filter.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const filterDefs = {
```
Comment (kenankigunda) [inline javascript/powercommands/task_client_filter.js]: Code structure: Consider cleaning these up using proper classes

File: javascript/powercommands/asana_power.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+const run = async () => {
+  try {
+    const projectId = await connectionSettings.getProjectId(argv.project, argv.save);
+    const token = await connectionSettings.getToken(argv.token, argv.save);
+    const command = commandRunner.getCommand(argv);
+    if (!command) return;
+    const filters = taskFilterer.getFilters(argv);
+    const client = await asana.Client.create().useAccessToken(token);
+    const tasks = await taskProvider.getTaskRefs(client, projectId);
+    tasks.stream().on('data', async taskRef => {
```
Comment (kenankigunda) [inline javascript/powercommands/asana_power.js]: Extension: Add a `limit` parameter to operate over only a certain number of tasks

File: javascript/powercommands/task_provider.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const getTaskRefs = async (client, projectId) => {
+  console.log('Retrieving all task references for project', projectId);
+  return await client.projects.tasks(projectId);
```
Comment (rossgrambo) [inline javascript/powercommands/task_provider.js]: I believe "await" here is unnecessary, as this request is returning a promise, and is currently being bound in a promise which is not doing any additional logic.

File: javascript/powercommands/task_provider.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const getTask = async (client, taskId) => {
+  console.log('Retrieving full task for task ID', taskId);
+  return await client.tasks.findById(taskId);
```
Comment (rossgrambo) [inline javascript/powercommands/task_provider.js]: Same as above

**Pull request merged**